README.md

```diff
@@ -199,7 +199,7 @@ Not to worry—`choice`s can have `optional` and `asymmetric` fields, just like
 
 An `optional` field of a `choice` must be paired with a fallback field, which is used as a backup in case the reader doesn't recognize or doesn't want to handle the original field. So readers aren't required to handle `optional` fields; hence, *optional*. Note that the fallback itself might be `optional`, in which case the fallback must have a fallback, etc. Eventually, the fallback chain ends with a required field. Readers will scan the fallback chain for the first field they recognize.
 
-*Note:* An `optional` field in a `choice` is not simply a field with a nullable type. The word "optional" here means that readers can ignore it and use a fallback instead, not that its payload might be missing.
+**Note:** An `optional` field in a `choice` is not simply a field with an [option type](https://en.wikipedia.org/wiki/Option_type) or [nullable type](https://en.wikipedia.org/wiki/Nullable_type). The word "optional" here means that readers can ignore it and use a fallback instead, not that its payload might be missing. It's tempting to assume things work the same way for `struct`s and `choice`s, but in reality things work in [dual](https://en.wikipedia.org/wiki/Dual_\(category_theory\)) ways: optionality for a `struct` relaxes the burden on writers (they don't have to set the field), whereas for a `choice` the burden is relaxed on readers (they don't have to handle the field). This important insight about duality comes from [category theory](https://en.wikipedia.org/wiki/Category_theory), a branch of mathematics which many people mistakenly think to be useless for programming.
```

Comment:
I'd remove "which many people mistakenly think to be useless for programming". I know you're passionate about it, but I think it has unnecessary emotional load that doesn't benefit most of the readers :)

Reply:
I came to the same conclusion a few minutes after merging this PR: https://github.com/stepchowfun/typical/pull/268